internal/util/dotnet_sdk_version.go

```diff
@@ -21,19 +24,34 @@ func DotnetSDKVersion(file string) (string, error) {
 		jsonStr, err = os.ReadFile(file)
 
 		if err != nil {
-			return "", err
+			return "", "", err
 		}
 	} else {
 		jsonStr = []byte(file)
 	}
 
 	json.Unmarshal(jsonStr, &sdkGlobal)
 
-	res, _ := NestedMapLookup(sdkGlobal, "sdk", "version")
+	version, _ := NestedMapLookup(sdkGlobal, "sdk", "version")
+	rollForward, _ := NestedMapLookup(sdkGlobal, "sdk", "rollForward")
 
-	result := ""
-	if res != nil {
-		result = fmt.Sprintf("%v", res)
+	versionStr := ""
+	if version != nil {
+		versionStr = fmt.Sprintf("%v", version)
 	}
-	return result, err
+
+	// if the follForward is not empty, then change the version constraint based on the string
+	if rollForward != nil {
+		switch rollForward {
+		case "latestPatch", "latestFeature":
+
+			// as this is the latest patch, remove the patch number from the version string
+			sv, _ := semver.NewVersion(versionStr)
```

Comment:
Dotnet versions aren't strictly semver, so you might need to handle it specially?
From: https://learn.microsoft.com/en-us/dotnet/core/tools/global-json
![Screenshot_2024-09-03-10-45-15-94_d0ef7a9810af20f7b394f4f927b723b0](https://github.com/user-attachments/assets/d12b4fa1-5d6f-4186-a02f-50dfbc6b0ebb)

Patch is only the last two digits of the semver 'patch' field, and feature is the first digit of the 'patch' field: `.znn`